drivers/wifi/nrf700x/zephyr/src/qspi/src/ficr_prog.c

```diff
@@ -224,140 +225,70 @@ int write_otp_memory(unsigned int otp_addr, unsigned int *write_val)
 		break;
 	case QSPI_KEY:
 		mask_val = QSPI_KEY_FLAG_MASK;
-		write_otp_location(QSPI_KEY, write_val[0]);
-		write_otp_location(QSPI_KEY+1, write_val[1]);
-		write_otp_location(QSPI_KEY+2, write_val[2]);
-		write_otp_location(QSPI_KEY+3, write_val[3]);
+		for (int i = 0; i < QSPI_KEY_LENGTH_BYTES / 4; i++) {
+			ret = write_otp_location(QSPI_KEY + i, write_val[i]);
+			if (ret == -ENOEXEC) {
+				LOG_ERR("QSPI_KEY:Update Exception\n");
+				goto _exit_otp_write;
+			}
+			LOG_INF("Written QSPI_KEY0 (0x%x) : 0x%04x\n",
+						(QSPI_KEY + i) << 2, write_val[i]);
+		}
 		write_otp_location(REGION_DEFAULTS, mask_val);
-
-		LOG_INF("Written QSPI_KEY0 (0x%x) : 0x%04x\n",
-						(QSPI_KEY+0) << 2, write_val[0]);
-		LOG_INF("Written QSPI_KEY1 (0x%x) : 0x%04x\n",
-						(QSPI_KEY+1) << 2, write_val[1]);
-		LOG_INF("Written QSPI_KEY2 (0x%x) : 0x%04x\n",
-						(QSPI_KEY+2) << 2, write_val[2]);
-		LOG_INF("Written QSPI_KEY3 (0x%x) : 0x%04x\n",
-						(QSPI_KEY+3) << 2, write_val[3]);
 		LOG_INF("Written REGION_DEFAULTS (0x%x) : 0x%04x\n",
 						(REGION_DEFAULTS) << 2, mask_val);
 		break;
 	case MAC0_ADDR:
 		mask_val = MAC0_ADDR_FLAG_MASK;
-		write_otp_location(MAC0_ADDR, write_val[0]);
-		write_otp_location(MAC0_ADDR+1, write_val[1]);
-		write_otp_location(REGION_DEFAULTS, mask_val);
 
+		for (int i = 0; i < 2; i++) {
+			ret = write_otp_location(MAC0_ADDR + i, write_val[i]);
+			if (ret == -ENOEXEC) {
+				LOG_ERR("MAC ADDR0:Update Exception\n");
+				goto _exit_otp_write;
+			}
+			LOG_INF("mac addr 0 : Reg%d (0x%x) = 0x%04x\n",
+						(i+1), (MAC0_ADDR + i) << 2, write_val[i]);
+		}
+
+		write_otp_location(REGION_DEFAULTS, mask_val);
 		LOG_INF("Written MAC address 0\n");
-		LOG_INF("mac addr 0 : Reg1 (0x%x) = 0x%04x\n",
-						(MAC0_ADDR) << 2, write_val[0]);
-		LOG_INF("mac addr 0 : Reg2 (0x%x) = 0x%04x\n",
-						(MAC0_ADDR+1) << 2, write_val[1]);
 		LOG_INF("Written REGION_DEFAULTS (0x%x) : 0x%04x\n",
 						(REGION_DEFAULTS) << 2, mask_val);
 		break;
 	case MAC1_ADDR:
 		mask_val = MAC1_ADDR_FLAG_MASK;
-		write_otp_location(MAC1_ADDR, write_val[0]);
-		write_otp_location(MAC1_ADDR+1, write_val[1]);
-		write_otp_location(REGION_DEFAULTS, mask_val);
 
+		for (int i = 0; i < 2; i++) {
```

Comment:
This could be a helper function

Reply:
Created a helper function for updating MAC address.